src/client/room.js

```diff
@@ -51,8 +51,20 @@ function RoomMorph(ide) {
         }
     };
 
-    // update on login
-    SnapCloud.onLogin = this.update.bind(this);
+    // update on login (changing room name if default)
+    SnapCloud.onLogin = function() {
+        myself.update.bind(myself);
```

Comment:
this should be `myself.update()`. This currently reads as simply creating a function but not invoking it